numba/core/interpreter.py

```diff
@@ -1283,6 +1423,29 @@ def op_JUMP_IF_FALSE_OR_POP(self, inst, pred):
     def op_JUMP_IF_TRUE_OR_POP(self, inst, pred):
         self._op_JUMP_IF(inst, pred=pred, iftrue=True)
 
+    def op_JUMP_IF_NOT_EXC_MATCH(self, inst, pred, tos, tos1):
+        truebr = inst.next
+        falsebr = inst.get_jump_target()
+        gv_fn = ir.Global(
+            "exception_match", eh.exception_match, loc=self.loc,
+        )
+        exc_match_name = '$exc_match'
+        self.store(value=gv_fn, name=exc_match_name, redefine=True)
+        # not wholly convinced by this as tos is seemingly None?!
```

Comment:
How are you seeing that TOS is None?

Reply:
When an exception is raised and the `except` handler is entered, (IIRC) it injects 6 stack values.